cmd/kubeadm/app/util/config/testdata/defaulting/node/defaulted.yaml

```diff
@@ -8,6 +8,7 @@ discovery:
   timeout: 5m0s
   tlsBootstrapToken: abcdef.0123456789abcdef
 kind: JoinConfiguration
+metadata: {}
```

Comment:
While do we need this one if it is omitempty/optional?

Reply:
Because of the way the JSON library serializes stuff. Effectively, `omitempty` has no meaning for embedded structs (even if they are empty).

Reply:
Uhmm I fear this is somehow related to the same problem that causes open API validation to add more info than necessary when using kubeadm config embedded in CABPK api.
We should make sure this is not backfiring us and our tentative to give cluster API user a better experience